dali/operators/image/remap/cvcuda/warp_perspective.cc

```diff
@@ -0,0 +1,213 @@
+      }
+    } else if (fill_value_arg_.size() == 1) {
+      auto fv = fill_value_arg_[0];
+      float4 fill_value{fv, fv, fv, fv};
+      return fill_value;
+    } else {
+      return float4{0, 0, 0, 0};
+    }
+  }
+
+  void ValidateTypes(const Workspace &ws) const {
+    auto inp_type = ws.Input<GPUBackend>(0).type();
+    DALI_ENFORCE(inp_type == DALI_UINT8 || inp_type == DALI_INT16 || inp_type == DALI_UINT16 ||
+                     inp_type == DALI_FLOAT,
+                 "The operator accepts the following input types: "
+                 "uint8, int16, uint16, float.");
+    if (ws.NumInput() > 1) {
+      auto mat_type = ws.Input<GPUBackend>(1).type();
+      DALI_ENFORCE(mat_type == DALI_FLOAT,
+                   "Transformation matrix can be provided only as float values");
+    }
+  }
+
+  bool SetupImpl(std::vector<OutputDesc> &output_desc, const Workspace &ws) override {
+    ValidateTypes(ws);
+    const auto &input = ws.Input<GPUBackend>(0);
+    auto input_shape = input.shape();
+    auto input_layout = input.GetLayout();
+    output_desc.resize(1);
+
+    auto output_shape = input_shape;
+    int channels = (input_layout.find('C') != -1) ? input_shape[0][input_layout.find('C')] : -1;
+    fill_value_ = GetFillValue(channels);
+    if (!shape_arg_.empty()) {
+      auto height = std::max<int>(std::roundf(shape_arg_[0]), 1);
+      auto width = std::max<int>(std::roundf(shape_arg_[1]), 1);
+      auto out_sample_shape = (channels != -1) ? TensorShape<>({height, width, channels}) :
+                                                 TensorShape<>({height, width});
+      output_shape = TensorListShape<>::make_uniform(input.num_samples(), out_sample_shape);
+    }
+
+    output_desc[0] = {output_shape, input.type()};
+    return true;
+  }
+
+  void RunImpl(Workspace &ws) override {
+    const auto &input = ws.Input<GPUBackend>(0);
+    auto &output = ws.Output<GPUBackend>(0);
+    output.SetLayout(input.GetLayout());
+
+    kernels::DynamicScratchpad scratchpad({}, AccessOrder(ws.stream()));
+
+    nvcv::Tensor matrix{};
+    if (ws.NumInput() > 1) {
+      DALI_ENFORCE(!matrix_arg_.HasExplicitValue(),
+                   "Matrix input and `matrix` argument should not be provided at the same time.");
+      Tensor<GPUBackend> matrix_tensor;
+      auto &matrix_input = ws.Input<GPUBackend>(1);
+      DALI_ENFORCE(is_uniform(matrix_input.shape()),
+                   "Matrix data has to be a uniformly shaped batch.");
+      if (matrix_input.IsDenseTensor()) {
+        matrix_tensor = const_cast<TensorList<GPUBackend> &>(matrix_input).AsTensor();
+      } else {
+        matrix_data_.Copy(matrix_input, AccessOrder(ws.stream()));
+        matrix_tensor = matrix_data_.AsTensor();
+      }
+      matrix = nvcvop::AsTensor(matrix_tensor, "NW");
+    } else {
+      matrix = AcquireTensorArgument(ws, scratchpad, matrix_arg_, TensorShape<1>(9),
+                                     nvcvop::GetDataType<float>(), "W");
```

Comment:
I'm pretty sure we need to apply a fixup to the matrix to match WarpPerspective. We can add OpenCV compatibility (here and in WarpAffine) as an option, but I think being self-consistent is far better than randomly matching a patchwork of common libraries.

Reply:
Done + added tests for compatibility with warp_affine